### PR TITLE
discord: fix issues with command-line usage

### LINKED ISSUE #5367: When not using firecfg launching discord from command line is prone to fail.
A user on our IRC channel (rahl-) brought to our attention that due to the multiple naming used in our discord profiles things can break rather easily, leaving users in a confused state.

<details>
 <summary> $ firejail /opt/discord/Discord </summary>

```
Originally pasted by rahl- on https://bpa.st/7YLA
Added here for clarity and archival purposes, as these pastes
tend to dissappear from pastebin websites rather quickly.

- - - - - -

[0] ~$ which discord
/usr/bin/discord

[0] ~$ ll $(which discord)
lrwxrwxrwx 1 root root 20 Aug 10 21:27 /usr/bin/discord -> /opt/discord/Discord

[0] ~$ ls /usr/local/bin/discord
ls: cannot access '/usr/local/bin/discord': No such file or directory

## The following works as expected
[0] ~$ firejail discord
Reading profile /etc/firejail/discord.profile
Reading profile /etc/firejail/discord-common.profile
Reading profile /etc/firejail/electron.profile
Reading profile /etc/firejail/disable-common.inc
Reading profile /etc/firejail/disable-devel.inc
Reading profile /etc/firejail/disable-exec.inc
Reading profile /etc/firejail/disable-programs.inc
Reading profile /etc/firejail/whitelist-common.inc
Reading profile /etc/firejail/whitelist-var-common.inc
Parent pid 1222171, child pid 1222172
Private /opt installed in 401.66 ms
16 programs installed in 48.38 ms
Warning: skipping alternatives for private /etc
Warning: skipping crypto-policies for private /etc
Warning: skipping ld.so.preload for private /etc
Warning: skipping password for private /etc
Warning: skipping pki for private /etc
Private /etc installed in 40.65 ms
Warning: skipping alternatives for private /usr/etc
Warning: skipping ca-certificates for private /usr/etc
Warning: skipping crypto-policies for private /usr/etc
Warning: skipping fonts for private /usr/etc
Warning: skipping group for private /usr/etc
Warning: skipping ld.so.cache for private /usr/etc
Warning: skipping ld.so.preload for private /usr/etc
Warning: skipping localtime for private /usr/etc
Warning: skipping login.defs for private /usr/etc
Warning: skipping machine-id for private /usr/etc
Warning: skipping password for private /usr/etc
Warning: skipping pki for private /usr/etc
Warning: skipping pulse for private /usr/etc
Warning: skipping resolv.conf for private /usr/etc
Warning: skipping ssl for private /usr/etc
Private /usr/etc installed in 0.23 ms
Warning: /sbin directory link was not blacklisted
Warning: /usr/sbin directory link was not blacklisted
Warning: not remounting /run/user/1000/gvfs
Child process initialized in 605.01 ms
Discord 0.0.19
Starting app.
Starting updater.
[Modules] Modules initializing
[Modules] Distribution: remote
[Modules] Host updates: enabled
[Modules] Module updates: enabled
[Modules] Module install path: /home/<user>/.config/discord/0.0.19/modules
[Modules] Module installed file path: /home/<user>/.config/discord/0.0.19/modules/installed.json
[Modules] Module download path: /home/<user>/.config/discord/0.0.19/modules/pending
[Modules] No updates to install
[Modules] Checking for host updates.
[Modules] Host is up to date.
[Modules] Checking for module updates at https://discord.com/api/modules/stable/versions.json
[Modules] No module updates available.
Module ./ElectronTestRpc was not included.
[Modules] Starting to install discord_krisp...
[Modules] Fetching discord_krisp@0 from https://discord.com/api/modules/stable/discord_krisp/0
[Modules] Streaming discord_krisp@0 to /home/<user>/.config/discord/0.0.19/modules/pending/discord_krisp-0.zip
[Modules] Checking for host updates.
Error downloading with electron net: HTTP Error: Status Code 403
Falling back to node net library..
/usr/local/bin/xdg-mime: line 998: awk: command not found
/usr/local/bin/xdg-mime: line 998: awk: command not found
[Modules] Host is up to date.
[Modules] Checking for module updates at https://discord.com/api/modules/stable/versions.json
[Modules] No module updates available.
/usr/local/bin/xdg-open: line 811: : command not found
/usr/local/bin/xdg-open: line 881: x-www-browser: command not found
/usr/local/bin/xdg-open: line 881: firefox: command not found
/usr/local/bin/xdg-open: line 881: iceweasel: command not found
/usr/local/bin/xdg-open: line 881: seamonkey: command not found
/usr/local/bin/xdg-open: line 881: mozilla: command not found
/usr/local/bin/xdg-open: line 881: epiphany: command not found
/usr/local/bin/xdg-open: line 881: konqueror: command not found
/usr/local/bin/xdg-open: line 881: chromium: command not found
/usr/local/bin/xdg-open: line 881: chromium-browser: command not found
/usr/local/bin/xdg-open: line 881: google-chrome: command not found
/usr/local/bin/xdg-open: line 881: www-browser: command not found
/usr/local/bin/xdg-open: line 881: links2: command not found
/usr/local/bin/xdg-open: line 881: elinks: command not found
/usr/local/bin/xdg-open: line 881: links: command not found
/usr/local/bin/xdg-open: line 881: lynx: command not found
/usr/local/bin/xdg-open: line 881: w3m: command not found
xdg-open: no method available for opening 'https://discord.com/handoff?rpc=6463&key=0c8c42d1-71c4-4363-b3ee-93294bf0fe0d'

Parent is shutting down, bye...

## The following also works as expected
[1] ~$ firejail /usr/bin/discord
Reading profile /etc/firejail/discord.profile
Reading profile /etc/firejail/discord-common.profile
Reading profile /etc/firejail/electron.profile
Reading profile /etc/firejail/disable-common.inc
Reading profile /etc/firejail/disable-devel.inc
Reading profile /etc/firejail/disable-exec.inc
Reading profile /etc/firejail/disable-programs.inc
Reading profile /etc/firejail/whitelist-common.inc
Reading profile /etc/firejail/whitelist-var-common.inc
Parent pid 1222601, child pid 1222602
Private /opt installed in 320.34 ms
16 programs installed in 37.30 ms
Warning: skipping alternatives for private /etc
Warning: skipping crypto-policies for private /etc
Warning: skipping ld.so.preload for private /etc
Warning: skipping password for private /etc
Warning: skipping pki for private /etc
Private /etc installed in 30.06 ms
Warning: skipping alternatives for private /usr/etc
Warning: skipping ca-certificates for private /usr/etc
Warning: skipping crypto-policies for private /usr/etc
Warning: skipping fonts for private /usr/etc
Warning: skipping group for private /usr/etc
Warning: skipping ld.so.cache for private /usr/etc
Warning: skipping ld.so.preload for private /usr/etc
Warning: skipping localtime for private /usr/etc
Warning: skipping login.defs for private /usr/etc
Warning: skipping machine-id for private /usr/etc
Warning: skipping password for private /usr/etc
Warning: skipping pki for private /usr/etc
Warning: skipping pulse for private /usr/etc
Warning: skipping resolv.conf for private /usr/etc
Warning: skipping ssl for private /usr/etc
Private /usr/etc installed in 0.17 ms
Warning: /sbin directory link was not blacklisted
Warning: /usr/sbin directory link was not blacklisted
Warning: not remounting /run/user/1000/gvfs
Child process initialized in 477.03 ms
Discord 0.0.19
Starting app.
Starting updater.
[Modules] Modules initializing
[Modules] Distribution: remote
[Modules] Host updates: enabled
[Modules] Module updates: enabled
[Modules] Module install path: /home/<user>/.config/discord/0.0.19/modules
[Modules] Module installed file path: /home/<user>/.config/discord/0.0.19/modules/installed.json
[Modules] Module download path: /home/<user>/.config/discord/0.0.19/modules/pending
[Modules] No updates to install
[Modules] Checking for host updates.
[Modules] Host is up to date.
[Modules] Checking for module updates at https://discord.com/api/modules/stable/versions.json
[Modules] No module updates available.
Module ./ElectronTestRpc was not included.
[Modules] Starting to install discord_krisp...
[Modules] Fetching discord_krisp@0 from https://discord.com/api/modules/stable/discord_krisp/0
[Modules] Streaming discord_krisp@0 to /home/<user>/.config/discord/0.0.19/modules/pending/discord_krisp-0.zip
[Modules] Checking for host updates.
Error downloading with electron net: HTTP Error: Status Code 403
Falling back to node net library..

Parent is shutting down, bye...

## The following fails
[0] ~$ firejail /opt/discord/Discord
Reading profile /etc/firejail/Discord.profile
Reading profile /etc/firejail/discord-common.profile
Reading profile /etc/firejail/electron.profile
Reading profile /etc/firejail/disable-common.inc
Reading profile /etc/firejail/disable-devel.inc
Reading profile /etc/firejail/disable-exec.inc
Reading profile /etc/firejail/disable-programs.inc
Reading profile /etc/firejail/whitelist-common.inc
Reading profile /etc/firejail/whitelist-var-common.inc
Parent pid 1222565, child pid 1222566
Warning: skipping Discord for private /opt
Private /opt installed in 0.07 ms
15 programs installed in 38.31 ms
Warning: skipping alternatives for private /etc
Warning: skipping crypto-policies for private /etc
Warning: skipping ld.so.preload for private /etc
Warning: skipping password for private /etc
Warning: skipping pki for private /etc
Private /etc installed in 28.46 ms
Warning: skipping alternatives for private /usr/etc
Warning: skipping ca-certificates for private /usr/etc
Warning: skipping crypto-policies for private /usr/etc
Warning: skipping fonts for private /usr/etc
Warning: skipping group for private /usr/etc
Warning: skipping ld.so.cache for private /usr/etc
Warning: skipping ld.so.preload for private /usr/etc
Warning: skipping localtime for private /usr/etc
Warning: skipping login.defs for private /usr/etc
Warning: skipping machine-id for private /usr/etc
Warning: skipping password for private /usr/etc
Warning: skipping pki for private /usr/etc
Warning: skipping pulse for private /usr/etc
Warning: skipping resolv.conf for private /usr/etc
Warning: skipping ssl for private /usr/etc
Private /usr/etc installed in 0.21 ms
Warning: /sbin directory link was not blacklisted
Warning: /usr/sbin directory link was not blacklisted
Warning: not remounting /run/user/1000/gvfs
Child process initialized in 170.56 ms
Error: no suitable /opt/discord/Discord executable found

Parent is shutting down, bye...

# Now we try with the script stub
[0] ~$ sudo mv /usr/local/bin/{.,}discord
[0] ~$ which discord
/usr/local/bin/discord

# Had an odd case in between where running `discord` did not run the script
[0] ~$ hash -r
[0] ~$ which discord
/usr/local/bin/discord

# The following fails - I'm wondering if this is somehow related to the earlier
# fail case
[0] ~$ discord
Reading profile /etc/firejail/discord.profile
Reading profile /etc/firejail/discord-common.profile
Reading profile /etc/firejail/electron.profile
Reading profile /etc/firejail/disable-common.inc
Reading profile /etc/firejail/disable-devel.inc
Reading profile /etc/firejail/disable-exec.inc
Reading profile /etc/firejail/disable-programs.inc
Reading profile /etc/firejail/whitelist-common.inc
Reading profile /etc/firejail/whitelist-var-common.inc
Parent pid 1223314, child pid 1223315
Private /opt installed in 397.17 ms
16 programs installed in 45.69 ms
Warning: skipping alternatives for private /etc
Warning: skipping crypto-policies for private /etc
Warning: skipping ld.so.preload for private /etc
Warning: skipping password for private /etc
Warning: skipping pki for private /etc
Private /etc installed in 36.88 ms
Warning: skipping alternatives for private /usr/etc
Warning: skipping ca-certificates for private /usr/etc
Warning: skipping crypto-policies for private /usr/etc
Warning: skipping fonts for private /usr/etc
Warning: skipping group for private /usr/etc
Warning: skipping ld.so.cache for private /usr/etc
Warning: skipping ld.so.preload for private /usr/etc
Warning: skipping localtime for private /usr/etc
Warning: skipping login.defs for private /usr/etc
Warning: skipping machine-id for private /usr/etc
Warning: skipping password for private /usr/etc
Warning: skipping pki for private /usr/etc
Warning: skipping pulse for private /usr/etc
Warning: skipping resolv.conf for private /usr/etc
Warning: skipping ssl for private /usr/etc
Private /usr/etc installed in 0.19 ms
Warning: /sbin directory link was not blacklisted
Warning: /usr/sbin directory link was not blacklisted
Warning: not remounting /run/user/1000/gvfs
Child process initialized in 571.03 ms
/usr/bin/discord: line 3: /usr/bin/firejail: No such file or directory

Parent is shutting down, bye...

```

 </details>

We can improve the profiles to catch these kind of situations and avoid future issues/questions from our users. I'll be opening a PR on this shortly. But we need to look at implementing a safe and reliable way to open URLs in whatever web browser the user is using, which is currently still broken. This relates to the below and probably many others:
- #3881
- #5160
- #5364

### PR DESCRIPTION
**Pull request by glitsj16**
This PR fixes #5367.

### CODE REVIEW

**Review by kmk3 — COMMENTED** [review]
Suggestion: Make the CamelCase profile redirect to the lisp-case profile.  This
seems to be what is done for other profiles with similar naming conventions,
such as Telegram.profile / telegram.profile and Logs.profile /
gnome-logs.profile (and some other profiles that also start with `gnome-`).

For example, make Discord.profile redirect to discord.profile and leave all the
`private-` options only on discord.profile (to avoid duplication).

**Comment by kmk3** [comment]
With regards to the CI checks, sort.py seems to ignore the casing (see
`casefold`), so the hyphen is probably the only thing that matters sort-wise in
this PR.

It looks like all checks were passing when putting all CamelCase items last, so
trying that again should work.

File: etc/profile-a-l/discord-common.profile [hunk 1]
```diff
@@ -23,7 +23,7 @@ ignore novideo
 whitelist ${HOME}/.config/BetterDiscord
 whitelist ${HOME}/.local/share/betterdiscordctl
 
-private-bin bash,cut,echo,egrep,electron,electron[0-9],electron[0-9][0-9],fish,grep,head,sed,sh,tclsh,tr,xdg-mime,xdg-open,zsh
```
Comment (reinerh) [inline etc/profile-a-l/discord-common.profile:26]: do you know why electron was in there before? are there users running something like `firejail --profile=discord.profile electron /opt/discord/...` ?

Comment (glitsj16) [reply etc/profile-a-l/discord-common.profile:26]: I _assume_ it's due to the way arch linux (and possibly other distro's) packages electron to support having different versions installed side-by-side. The latest electron package always uses the shell script as /usr/bin/electron, while older electrons install it as a versioned /usr/bin/electronX.

Comment (kmk3) [reply etc/profile-a-l/discord-common.profile:26]: @reinerh [on Sep 14](https://github.com/netblue30/firejail/pull/5368#discussion_r970497695):

> do you know why electron was in there before? are there users running
> something like `firejail --profile=discord.profile electron /opt/discord/...`
> ?

For reference, it was added on commit ed1f0c088 ("Add electron[0-9]{0,2} to all
electron redirect …", 2021-08-01) by @rusty-snake.

Presumably, electron-based programs need to execute the electron binary
directly.

**Review by kmk3 — APPROVED** [review]
LGTM.

**Review by reinerh — APPROVED** [review]

**Pull request merged**